pyccel/parser/syntactic.py

```diff
@@ -729,13 +730,18 @@ def fill_types(ls):
 
                 txt  = '#$ header template ' + str(tp_name)
                 txt += '(' + '|'.join(types) + ')'
-                if tp_name in templates:
+                if tp_name in tmp_templates:
                     msg = 'The template "{}" is duplicated'.format(tp_name)
                     errors.report(msg,
                                 bounding_box = (stmt.lineno, stmt.col_offset),
                                 severity='warning')
+                tmp_templates[tp_name] = hdr_parse(stmts=txt)
+                template['templates_dict'] = tmp_templates
+            template['template_list'] = decorators['template']
+            decorators['template'] = template
```

Comment:
Could you explain the purpose of `template_list` please. Will we have something like:
```
decorators = {
...
template : {'template_list':[FunctionCall('template','T',('int32', 'int64')),
                             FunctionCall('template','S','('int','float'))],
            'templates_dict':{'T':Template(...), 'S', Template(...)
           }
}
```

Reply:
I have had another look at `pycode.py` and I think I understand the purpose. Is this done so that we can retrieve the templates in decorator format for printing decorators?

If so could you change the names to reflect this. E.g. `tempate_dict`, `decorator_list`